### PR TITLE
Scott/358 feature add pegout witness data to edh

### PR DESCRIPTION
**Pull request by scottmillner**

### CODE REVIEW

File: crates/botanix-lib/src/extra_data_header.rs [hunk 1]
```diff
@@ -308,6 +342,14 @@ mod tests {
         let pk = secp256k1::PublicKey::from_slice(&maybe_pk.as_slice()).expect("a public key");
         // Check the public key is the same as one provided
         assert_eq!(pk, public_key);
+        // Check the length of the witness data
+        assert_eq!(buf[74..78], vec![1u8, 0u8, 0u8, 0u8].as_slice().to_owned());
+        // Check the witness data
+        let binding = buf[78..].to_vec();
+        let mut maybe_witness = binding.as_slice();
```
Comment (ozymandias-rex) [inline crates/botanix-lib/src/extra_data_header.rs:349]: nit: does this have to be mutable ?

**Review by ozymandias-rex — APPROVED** [review]

File: crates/consensus/authority/src/lib.rs [hunk 2]
```diff
@@ -344,12 +345,19 @@ where
             .unwrap();
         header.state_root = state_root;
 
+        let witness_data_final = if header.is_poa_epoch() {
+            witness_data.clone().map_or(Some(vec![]), |witness_data| Some(witness_data))
```
Comment (arminsabouri) [inline crates/consensus/authority/src/lib.rs:350]: Maybe leave a TODO here that if there are pegouts pending from this epoch and there are no witnesses then we need to fail consensus validation

**Review by arminsabouri — COMMENTED** [review]
Approach ACK

File: crates/consensus/authority/src/lib.rs (hunk 2 above)
Comment (scottmillner) [reply crates/consensus/authority/src/lib.rs:350]: todo added

File: crates/botanix-lib/src/extra_data_header.rs (hunk 1 above)
Comment (scottmillner) [reply crates/botanix-lib/src/extra_data_header.rs:349]: won't compile without mut in both places

**Pull request merged**